sdk/cosmos/azure-cosmos/azure/cosmos/_cosmos_http_logging_policy.py

```diff
@@ -0,0 +1,135 @@
+import os
+import urllib
+
+from azure.core.pipeline.policies import HttpLoggingPolicy
+import logging
+import types
+
+
+
+class CosmosHttpLoggingPolicy(HttpLoggingPolicy):
+
+    def __init__(self, logger=None, enable_diagnostics_logging=False, **kwargs):
+        self._diagnostics_mode = enable_diagnostics_logging
```

Comment:
This mechanism should not be a mode - the existing policy should always behave consistently.
Rather, if `enable_diagnostics_logging` is set, _additional_ behaviour should be applied.